drivers/flash/flash_stm32l5_u5.c

```diff
@@ -356,50 +358,52 @@ void flash_stm32_page_layout(const struct device *dev,
 {
 	FLASH_TypeDef *regs = FLASH_STM32_REGS(dev);
 	static struct flash_pages_layout stm32_flash_layout[3];
-#define PAGES_PER_BANK ((FLASH_SIZE / FLASH_PAGE_SIZE) / 2)
+
+	*layout = stm32_flash_layout;
+	*layout_size = ARRAY_SIZE(stm32_flash_layout);
+
+	if (stm32_flash_layout[0].pages_count != 0) {
+		/* Short circuit calculation logic if already performed */
+		return;
+	}
```

Comment:
Are you sure about this part ?
`stm32_flash_layout` is local to this function, so in order for this check to work, `stm32_flash_layout` should be declared outside of this function and zero initialized

Reply:
ok for the zero initiliazed part, but the variable now has to be global, otherwise I don't see how this could work (since you actually changed the logic vs previous version, now check is done on a zero initialized variable)

Reply:
@erwango I'm not sure I'm following you.

`stm32_flash_layout[]` has not changed at all in this commit. Being a static local, it was and will continue to be zero initialized on application startup.

What has changed is that the duplicate checks from old (-) lines 367 and 387 (see below if you're looking at the "files changed" tab on github.com) have been combined into the check in new (+) line 365. The statement logic has been inverted so an early return can be used, but the control flow is the same.

The other change is moving the assignment of `*layout` and `*layout_size` to the start of the function body instead of at the end. Again, this is to facilitate the early return. There is no difference in the control flow.

Reply:
Just in case your concern is about `stm32_flash_layout[]` going out of scope and becoming garbage on the stack, because it is declared static, it is not placed on the stack and instead is placed in .bss. Therefore, my understanding is that its lifetime is forever and this behavior is fine.

Reply:
That was my concern indeed. Thanks for clarifying.